### ISSUE TITLE
[CI] UpgradeClusterClientYamlTestSuiteIT failure in suite :x-pack:qa:rolling-upgrade:v7.6.0#twoThirdsUpgradedTest

### ISSUE THREAD

**Issue by astefan** [elastic/elasticsearch#75374]
**Build scan**: https://gradle-enterprise.elastic.co/s/xazhuw2lv4xok

**Repro line**:
```
gradlew ':x-pack:qa:rolling-upgrade:v7.6.0#twoThirdsUpgradedTest' -Dtests.class="org.elasticsearch.upgrades.UpgradeClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=mixed_cluster/80_transform_jobs_crud/Test GET, start, and stop old cluster batch transforms}" -Dtests.seed=B919838B66982C1D -Dtests.bwc=true -Dtests.locale=zh-HK -Dtests.timezone=Africa/Conakry -Druntime.java=8
```

**Reproduces locally?**:  yes, but the assertion error provided below is not in the nodes' logs

**Applicable branches**: 7.x

**Failure history**:
There were some failures starting yesterday, but the assertion failure is not there.

**Failure excerpt**:

```
09:23:51 » [2021-07-15T06:16:09,351][ERROR][o.e.b.ElasticsearchUncaughtExceptionHandler] [v7.6.0-0] fatal error in thread [Thread-7], exiting
09:23:51 »  java.lang.AssertionError: org.elasticsearch.search.internal.ShardSearchRequest/eJ60j0jESEa9UZLXUQcC2w:511
09:23:51 »  	at org.elasticsearch.search.SearchService.getShard(SearchService.java:431) ~[elasticsearch-7.15.0-SNAPSHOT.jar:7.15.0-SNAPSHOT]
09:23:51 »  	at org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:405) ~[elasticsearch-7.15.0-SNAPSHOT.jar:7.15.0-SNAPSHOT]
09:23:51 »  	at org.elasticsearch.action.search.SearchTransportService.lambda$registerRequestHandler$6(SearchTransportService.java:311) ~[elasticsearch-7.15.0-SNAPSHOT.jar:7.15.0-SNAPSHOT]
09:23:51 »  	at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$ProfileSecuredRequestHandler$1.doRun(SecurityServerTransportInterceptor.java:261) ~[?:?]
09:23:51 »  	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26) ~[elasticsearch-7.15.0-SNAPSHOT.jar:7.15.0-SNAPSHOT]
09:23:51 »  	at org.elasticsearch.common.util.concurrent.EsExecutors$DirectExecutorService.execute(EsExecutors.java:228) ~[elasticsearch-7.15.0-SNAPSHOT.jar:7.15.0-SNAPSHOT]
09:23:51 »  	at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$ProfileSecuredRequestHandler.lambda$messageReceived$0(SecurityServerTransportInterceptor.java:315) ~[?:?]
09:23:51 »  	at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:134) ~[elasticsearch-7.15.0-SNAPSHOT.jar:7.15.0-SNAPSHOT]
```

It seems there are some incompatibilities between 7.x and 7.6 with the above assertion failing.
Some tests are failing because the test suite times out, and some fail because one of the nodes is not available, but the assertion error above looks troubling nonetheless.

**Comment by astefan** [comment]
Found another identical failure in another run: https://gradle-enterprise.elastic.co/s/ipagefwpjmiru

**Comment by dnhatn** [comment]
I think this is caused by #75333. I am looking at that backport PR.

**Comment by dnhatn** [comment]
Fixed in #75403
